site/content/en/docs/concepts/resource_flavor.md

```diff
@@ -3,110 +3,90 @@ title: "Resource Flavor"
-   cannot be assigned to a Workload's podSet.
+1. When admitting a Workload, Kueue evaluates the [`.nodeSelector`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) and [`.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) fields in the PodSpecs of your [Workload](/docs/concepts/workload) against the ResourceFlavor labels.
+`ResourceFlavors` that don't match the node affinity of the Workload cannot be assigned to a Workload's podSet.
 
 
 2. Once the Workload is admitted:
-   - Kueue adds the ResourceFlavor labels to the
-    `.nodeSelector` of the underlying Workload Pod templates. This occurs if the Workload
-     didn't specify the `ResourceFlavor` labels already as part of its nodeSelector.
-
-     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/),
-     Kueue adds the labels to the `.spec.template.spec.nodeSelector` field. This
-     guarantees that the workload's Pods can only be scheduled on the nodes
-     targeted by the flavor that Kueue assigned to the Workload.
-   
+   - Kueue adds the ResourceFlavor labels to the `.nodeSelector` of the underlying Workload Pod templates.
+   This occurs if the Workload didn't specify the `ResourceFlavor` labels already as part of its nodeSelector.
+
+     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/), Kueue adds the labels to the `.spec.template.spec.nodeSelector` field.
+     This guarantees that the workload's Pods can only be scheduled on the nodes targeted by the flavor that Kueue assigned to the Workload.
+
    - Kueue adds the tolerations to the underlying Workload Pod templates.
 
-     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/),
-     Kueue adds the tolerations to the `.spec.template.spec.tolerations` field. This allows that the 
-     workloads Pods to be scheduled on nodes having specific taints.
+     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/), Kueue adds the tolerations to the `.spec.template.spec.tolerations` field.
+     This allows that the workloads Pods to be scheduled on nodes having specific taints.
 
-## ResourceFlavor taints
+A sample ResourceFlavor of this type looks like the following:
 
-To restrict the usage of a ResourceFlavor, you can configure the `.spec.nodeTaints` field.
-These taints should typically match the taints of the Nodes associated with the ResourceFlavor.
+{{< include "examples/admin/resource-flavor-tolerations.yaml" "yaml" >}}
+
+When defining a ResourceFlavor as above, you can set the following values:
+- The `.metadata.name` field to reference a ResourceFlavor from a [ClusterQueue](/docs/concepts/cluster_queue) in the `.spec.resourceGroups[*].flavors[*].name` field.
```

Comment:
metadata.name is required, but it sounds optional here (and below in the resource-flavor-taints section).

Perhaps add something about how name is used to uniquely identify a ResourceFlavor (as you did with ClusterQueue)